risc0/binfmt/src/image.rs

```diff
@@ -213,44 +225,48 @@ impl MemoryImage {
     /// Reads the given byte array in this memory image at the given
     /// address  The caller is responsible for ensuring the bytes do
     /// not overlap a page boundry.
-    pub fn load_region_in_page(&self, addr: u32, bytes: &mut [u8]) {
+    pub fn load_region_in_page(&self, addr: u32, bytes: &mut [u8]) -> Result<()> {
         let page_idx = self.info.get_page_index(addr);
         let page_start = self.info.get_page_addr(page_idx);
 
-        if let Some(page) = self.pages.get(&page_idx) {
-            bytes.clone_from_slice(
-                &page[(addr - page_start) as usize..(addr - page_start) as usize + bytes.len()],
-            );
-        } else {
-            assert!(
-                addr as usize <= MEM_SIZE,
-                "address {addr:08X} outside MEM_SIZE ({MEM_SIZE:08X})"
-            );
-            bytes.fill(0);
-        }
+        match self.pages.get(&page_idx) {
```

Comment:
Personally, I prefer `if let Some` over `match`

Reply:
Reverted in #1226 👍